core/shared/src/main/scala/zio/internal/FiberRuntime.scala

```diff
@@ -244,7 +244,7 @@ final class FiberRuntime[E, A](fiberId: FiberId.Runtime, fiberRefs0: FiberRefs,
 
       while (evaluationSignal == EvaluationSignal.Continue) {
         evaluationSignal = {
-          val message = inbox.poll()
+          val message = inbox.relaxedPoll()
```

Comment:
I'm not 1000% sold about using `relaxedPoll` here. Which benchmarks did you notice an improvement for by using `relaxedPoll` here?

Reply:
In my latest iteration I have removed the use of relaxedPoll because it seems to only benefit synthetic workloads. In real-world usage it seems the use of the `spinWait` likely puts us ahead in the small window where the queue may not be empty but the message is not yet visible to the consumer.